agent/src/agentclient.rs

```diff
@@ -165,6 +165,8 @@ impl<T: APIServerClient> BrupopAgent<T> {
     async fn gather_system_metadata(
```

Comment:
Nit: I think this function name may be showing that it was a strange choice. Maybe something like `current_shadow_status` would be clearer? I don't exactly love that name either. There may be something better.

Reply:
I like that suggestion!